### PR TITLE
WIP - Feat/airflow emreks

### PR DESCRIPTION
**Pull request by srinivasreddych**
*Issue #, if available:*

- Added a module to support the execution of Spark Jobs on AWS EMR on EKS.
- Added an example spark dags module to demonstrate how to consume `emr-on-eks` module for running spark jobs on AWS EMR on EKS

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: modules/simulations/airflow-emreks/README.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  - name: raw-bucket-name
+    valueFrom:
+      moduleMetadata:
+        group: optionals
+        name: datalake-buckets
+        key: RawBucketName
+```
+
+#### Output Example
+
```
Comment (dgraeber) [inline modules/simulations/airflow-emreks/README.md:88]: What are the output metadata elements that are written to SSM?  What would I expect to get back from `seedfarmer list moduledata` ?  This is difficult to understand

File: modules/simulations/airflow-emreks/README.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+For custom use-cases, you shuld be adjusting the required permissions on EMR Job execution role.
+
+## Inputs/Outputs
+
+
+### Input Parameters
+
+```yaml
+name: emr-on-eks
```
Comment (dgraeber) [inline modules/simulations/airflow-emreks/README.md:14]: Can you define the actual input and output parameters as well as add the examples....

File: modules/simulations/airflow-emreks/README.md (hunk 1 above)
Comment (srinivasreddych) [reply modules/simulations/airflow-emreks/README.md:88]: Added

**Review by dgraeber — DISMISSED** [review]

File: modules/simulations/airflow-emreks/airflow_emr_eks.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+from aws_cdk import aws_emrcontainers as emrc
+from aws_cdk import aws_iam as iam
+
+# from cdk_nag import NagSuppressions
+from constructs import Construct, IConstruct
+
+"""
+This stack deploys the following:
+- EMR on EKS virtual cluster
+- Airflow with EMR on EKS
```
Comment (kevinsoucy) [inline modules/simulations/airflow-emreks/airflow_emr_eks.py:20]: This is deploying "Airflow DAGs using EMR on EKS" not MWAA, right?

**Comment by kevinsoucy** [comment]
Hey @srinivasreddych  - this looks really good, glad you got this working. I think it would make sense to split this module up into 2 modules,  by moving the emr on eks stacks to a core/emr-on-eks module so this is more reusable. Then the individual use cases can have separate modules containing business logic including spark scripts, Airflow Dags, and IAM policies for the Spark Job. I am trying to leverage this in the ros-image-pipeline, but I am ending up copying and pasting a lot of your code to make use of it.

**Comment by JunjieTang-D1** [comment]
hi @kevinsoucy  when you refactor the ros-image-pipeline, can you create a new one with emr on EKS and keep the existing solution with EMR?

**Comment by kevinsoucy** [comment]
This looks good to me! Thanks @srinivasreddych !

File: modules/simulations/airflow-emreks/airflow_emr_eks.py (hunk 3 above)
Comment (srinivasreddych) [reply modules/simulations/airflow-emreks/airflow_emr_eks.py]: Clarified in the readme

File: modules/simulations/airflow-emreks/README.md (hunk 2 above)
Comment (srinivasreddych) [reply modules/simulations/airflow-emreks/README.md]: done

**Pull request merged**